src/rust/engine/ui/src/console_ui.rs

```diff
@@ -45,6 +46,8 @@ pub struct ConsoleUI {
   local_parallelism: usize,
   // While the UI is running, there will be an Instance present.
   instance: Option<Instance>,
+  teardown_in_progress: bool,
+  teardown_mpsc: (mpsc::Sender<()>, mpsc::Receiver<()>),
```

Comment:
Why not use the tokio channels? And if this is only a single value to be sent, why not a tokio `oneshot` channel?

Reply:
cf. https://docs.rs/tokio/0.2.22/tokio/sync/mpsc/index.html#communicating-between-sync-and-async-code, "Unbounded channel: You should use the kind of channel that matches where the receiver is. So for sending a message from async to sync, you should use the standard library unbounded channel or crossbeam. Similarly, for sending a message from sync to async, you should use an unbounded Tokio mpsc channel."

The receiver here is synchronous code, so the mpsc channel should also be the synchronous std lib one. I don't think there's a synchronous equivalent of tokio `oneshot`.